### DISCUSSION TITLE
bind 함수 내부에서 CollectionView 관련 기능이 작동하지 않는 문제

### DISCUSSION THREAD

**Discussion by nomatterjun** [Favor-Gift-Reminder/Favor-iOS#61] [Q&A] [answered]
음.. 저번에 한 번 가볍게 `collectionView.rx.itemSelected`가 동작하지 않는다고 했었잖아...?

이번에 마이페이지 화면 구현하면서 `scrollView`의 `contentOffset` 프로퍼티를 사용해야해서

https://github.com/Favor-Gift-Reminder/Favor-iOS/blob/1de431623b6bde11a72d508024cb2d9c71fd95f8/Favor/Favor/Sources/Scenes/MyPage/ViewControllers/MyPageVC.swift#L131-L137

이런식으로 넣어줬는데 동작을 안하더라고..?

리액터쪽은 이렇게 되어있어..

https://github.com/Favor-Gift-Reminder/Favor-iOS/blob/1de431623b6bde11a72d508024cb2d9c71fd95f8/Favor/Favor/Sources/Scenes/MyPage/Reactor/MyPageReactor.swift#L14-L63

별 특별할거 없는 코드인데..
컬렉션뷰나 테이블뷰 관련된 기능들만 동작하지 않는 것 같아

`setDelegate`도 시도해봤었는데 소용 없었고

[리액터킷 이슈](https://github.com/ReactorKit/ReactorKit/issues/93)에 같은 문제를 한 분이 올리셨던게 있긴 했는데
내 경우엔 `reactor`는 제대로 주입된걸 확인까지 했는데도 문제가 있는 상황이야

`bind` 함수는 제대로 동작하고 있어
`tap` 메서드 같은 동작들은 제대로 작동하고 `section`도 제대로 넣어지고 있으니깐..

혹시 관련해서 알고 있는 내용 있을까..?

> 콘솔 출력 내용

![스크린샷 2023-02-17 15 45 47](https://user-images.githubusercontent.com/60438045/219569734-5b7112cf-c7e2-426d-8cd8-84e80ed058f9.png)

**Comment by EnnzoKim** [comment, marked as answer]
헉 이걸 지금 봤네..
혹시 아직도 미해결된 문제일까?

**Reply by nomatterjun** [reply]
응..

 ``` swift
private func setupCollectionView() {
  Observable.just([])
    .bind(to: self.collectionView.rx.items(dataSource: self.dataSource))
    .disposed(by: self.disposeBag)

  self.collectionView.rx.contentOffset
    .skip(1)
    .bind(with: self, onNext: { owner, offset in
      owner.updateHeaderConstraintAndOpacity(offset: offset)
    })
    .disposed(by: self.disposeBag)
}
```

해결을 결국 못해서
일단 이렇게 따로 `bind`에서 빼서 따로 사용하고 있어..!

**Reply by EnnzoKim** [reply]
음 뭔가 
`setupCollectionView`에서는 작동이 되는데
`bind`에서는 작동이 되지 않는 것 보니까

나도 정확한 원인은 모르지만
`setupCollectionView`는 `viewDidLoad`에서는 호출되고
`bind`는 `viewDidLoad`이전에 호출이 되니까 
`collectionView`가 `self.view`에 로드되는 시점으로 인해 생기는 버그같아

이건 지금 형이 구현한대로 사용하는게 좋을 것 같아!
의견 부탁해~
